beta-pipelines/multiome/atac.wdl

```diff
@@ -43,125 +49,170 @@ workflow ATAC {
-      gzip ~{fastq_barcodes_read1} ~{fastq_barcodes_read3}
-      echo "These are the zipped files and sizes."
-      ls -l
-     >>>
+  input {
+    Array[File] read1_fastq
+    Array[File] read3_fastq
+    Array[File] barcodes_fastq
+    String read_structure = "16C"
+    String barcode_orientation = "FIRST_BP_RC"
+    String output_base_name
+    File whitelist
+
+    # [?] copied from corresponding optimus wdl for fastqprocessing 
+    # using the latest build of warp-tools in GCR
+    String docker = "us.gcr.io/broad-gotc-prod/warp-tools:1.0.1-1682971351"
+    # Runtime attributes [?]
+    Int mem_size = 5
+    Int cpu = 16   
+    # TODO decided cpu
+    # estimate that bam is approximately equal in size to fastq, add 20% buffer
+    Int disk_size = ceil(2 * ( size(read1_fastq, "GiB") + size(read3_fastq, "GiB") + size(barcodes_fastq, "GiB") )) + 400
+    Int preemptible = 3
+   }
 
-    # use docker image for given tool cutadapat
-    runtime {
-      docker: docker_image
-      disks: "local-disk ${disk_size} HDD"
-      memory: "${mem_size} GiB"
-    }
+  meta {
+    description: "Converts a set of fastq files to unaligned bam file/fastq file, also corrects barcodes and partitions the alignments by barcodes. Allows for variable barcode and umi lengths as input, if applicable."
+   }
 
-    output {
-      File fastq_barcodes_output_read1 = "~{fastq_barcodes_read1}.gz"
-      File fastq_barcodes_output_read3 = "~{fastq_barcodes_read3}.gz"
-    }
-  }
-  # trim read 1 and read 2 adapter sequeunce with cutadapt
-  task TrimAdapters {
+  parameter_meta {
+        read1_fastq: "Array of read 1 FASTQ files of paired reads -- forward reads"
+        read3_fastq: "Array of read 3 FASTQ files of paired reads -- reverse reads"
+        barcodes_fastq: "Array of read 2 FASTQ files which contains the cellular barcodes"
+        output_base_name: "Name of sample matching this file, inserted into read group header"
+        read_structure: "A string that specifies the barcode (C) positions in the Read 2 fastq"
+        barcode_orientation: "A string that specifies the orientation of barcode needed for scATAC data. The default is FIRST_BP. Other options include LAST_BP, FIRST_BP_RC or LAST_BP_RC."
+        whitelist: "10x genomics cell barcode whitelist for scATAC"
+        docker: "(optional) the docker image containing the runtime environment for this task"
+        mem_size: "(optional) the amount of memory (MiB) to provision for this task"
+        cpu: "(optional) the number of cpus to provision for this task"
+        disk_size: "(optional) the amount of disk space (GiB) to provision for this task"
+        preemptible: "(optional) if non-zero, request a pre-emptible instance and allow for this number of preemptions before running the task on a non preemptible machine"
+   }
+
+  command <<<
+  
+    set -euo pipefail
+    
+    # Cat files for each r1, r3 and barcodes together [? is this a good idea or should we just do what optimus does]
```

Comment:
"[? is this a good idea or should we just do what optimus does]" is this still an open question?

Reply:
I just added this as a comment for us for later improvements we might want to do. We can definitely remove the comment and just keep note of it for later on.